src/lib/agent.ts

```diff
@@ -62,13 +69,71 @@ export class Agent {
     eventbus.on("didGenerated", callback)
   }
 
+  private handleDiscoverFeatures(message: IMessage) {
+    const regexEscape = (s: string) => s.replace(/([.*+?$^=!:{}()|\[\]\/\\])/g, "\\$1");
+    const createRegex = (query: string) => (new RegExp(`^${query.split("*").map(regexEscape).join(".*")}$`));
+    let protocolResponse: object[] = [];
+
+    // Loop through all queries, then all implemented protocols and build up a
+    // list of supported protocols that match the user's request
+    for(let query of message.body.queries) {
+
+      // Rudimentary implementation, ignoring all except protocol requests
+      if(query["feature-type"] != "protocol")
+        continue
+
+      for(let protocol of IMPLEMENTED_MESSAGE_TYPES) {
+        if(createRegex(query["match"]).test(protocol)) {
+          protocolResponse.push({
+            "feature-type": "protocol",
+            "id": protocol,
```

Comment:
Example from the discover features spec:

```json
{
    "type": "https://didcomm.org/discover-features/2.0/disclose",
    "thid": "yWd8wfYzhmuXX3hmLNaV5bVbAjbWaU",
    "body":{
        "disclosures": [
            {
                "feature-type": "protocol",
                "id": "https://didcomm.org/tictactoe/1.0",
                "roles": ["player"]
            },
            {
                "feature-type": "goal-code",
                "id": "org.didcomm.sell.goods.consumer"
            }
        ]
    }
}
```

The `protocol` in the disclose message will be the Protocol Identifier URI -- as implemented, this will return a Message Type URI. So we would send `https://didcomm.org/basicmessage/2.0/message` when we should send `https://didcomm.org/basicmessage/2.0`.

Reply:
Good catch! Definitely something I missed